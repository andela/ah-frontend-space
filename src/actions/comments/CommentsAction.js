import axios from 'axios';

export const loading = () => ({ type: 'LOADING' });

export const loadComment = payload => ({ type: 'LOAD_COMMENT', payload });

export const failed = payload => ({ type: 'FAILED', payload });


export const getComments = slug => async dispatch => axios.get(
  `https://ah-haven-space.herokuapp.com/api/articles/${slug}/comments/`,
)
  .then(
    (res) => {
      dispatch(loadComment(res.data));
    },
  )
  .catch((error) => {
    dispatch(failed(error));
    console.log(error);
  });


export default getComments;
