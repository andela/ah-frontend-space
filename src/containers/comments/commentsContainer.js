
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import commentActions from '../../actions/comments/CommentsAction';
import Comments from '../../components/HomePage/CommentsComponent';


export class Comment extends Component {
  componentDidMount() {
    const { slug, fetchComments } = this.props;
    fetchComments(slug);
  }

  render() {
    const { comment: { comments } } = this.props;
    return (<Comments comments={comments} />);
  }
}


export const mapStateToProps = state => (
  {
    comment: state.comments.comment,
  }
);


Comment.propTypes = {
  comment: PropTypes.shape().isRequired,
  slug: PropTypes.string.isRequired,
  fetchComments: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, { fetchComments: commentActions })(Comment);
