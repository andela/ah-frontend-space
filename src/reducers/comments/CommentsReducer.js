const initialState = {
  loading: false,
  comment: '',
  error: '',
};


const commentReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'STARTED':
      return {
        ...state,
        loading: true,
      };

    case 'FAILED':
      return {
        ...state,
        ...action.payload,
      };

    case 'LOAD_COMMENT':
      return ({
        ...state,
        ...action.payload,
      });

    default:
      return state;
  }
};


export default commentReducer;
