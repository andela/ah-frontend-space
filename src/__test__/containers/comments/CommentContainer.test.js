import React from 'react';
import { shallow } from 'enzyme';
import { Comment, mapStateToProps } from '../../../containers/comments/commentsContainer';

describe.only('<Comment />', () => {
  let wrapper;
  const props = {
    comment: {},
    fetchComments: jest.fn(),
    slug: 'drew',
  };
  beforeEach(() => {
    wrapper = shallow(<Comment {...props} />);
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should map the comments reducer to props', () => {
    const state = {
      comments: {
        comment: '',
      },
    };

    expect(mapStateToProps(state)).toEqual({
      comment: '',

    });
  });
});
