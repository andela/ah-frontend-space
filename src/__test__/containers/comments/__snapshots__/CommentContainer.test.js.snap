// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment /> should match the snapshot 1`] = `ShallowWrapper {}`;
